libgnucash/engine/SX-ttinfo.hpp

```diff
@@ -31,9 +31,13 @@
 #include "Account.h"
 #include "gnc-commodity.h"
 
+#include <vector>
+
 typedef struct TTInfo_s TTInfo;
 typedef struct TTSplitInfo_s TTSplitInfo;
 
+using TTSplitInfoVec = std::vector<TTSplitInfo*>;
```

Comment:
`std::vector<TTSplitInfo>`. That way the std::vector will take care of managing the memory for you and TTInfo won't need a custom destructor. You can `reserve(2)` in the TTInfo constructor to avoid an expensive reallocation for the vast majority of cases; `reserve(3)` will waste a little memory and cover the investment cases with an empty tracker split. You could test for trading accounts and bump it up 2 more.

Reply:
> `std::vector<TTSplitInfo>`. That way the std::vector will take care of managing the memory for you and TTInfo won't need a custom destructor. You can `reserve(2)` in the TTInfo constructor to avoid an expensive reallocation for the vast majority of cases; `reserve(3)` will waste a little memory and cover the investment cases with an empty tracker split. You could test for trading accounts and bump it up 2 more.

How about this branch is reviewed for correctness and the auto memory management postponed to a later branch?

Reply:
[Correctness](https://en.wikipedia.org/wiki/Correctness_(computer_science)) is not somewhere you want to go. 

> This is mainly a simple file to practise c++ structs... and I'm not sure it's safe to use auto memory management

If you're going to practice then you should practice doing it the [right way](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rr-newdelete). Automatic memory management exists because manual memory management is inherently unsafe.